### DISCUSSION TITLE
Welcome to oi-userland Discussions!

### DISCUSSION THREAD

**Discussion by AndWac** [OpenIndiana/oi-userland#6285] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by DrLou** [comment]
Hello Friends!
What a great idea - to have the discussions here on GitHub...
Introduction? Well... I go back to the Early Days... Let's start at SunOS and Solaris... Then move to first iterations of OpenSolaris - in fact, even to OpenSolaris 'Indiana'.  Heck, I remember being wow'd at the meeting in New York when Sun Engineers were introducing a wacky thing called 'CrossBow'...  Add ZFS, and we were at _**The Pivot Point**_.
Was among the first adopters of OpenIndiana, and have never waned. Was it version 134a?
Have played various 'roles', starting with early adopter/enthusiast/evangelist - to experimenter/packager/other...

**Comment by Toasterson** [comment]
Hello everyone,

Good to see some discussions starting up. Let's keep this going to gather new people!

I am a happy contributor to OI and love illumos as a whole! Let's see what we can do together!

**Comment by jimklimov** [comment]
I do systems things, figure out neat tricks and help those who want it. Sadly, too lazy and overwhelmed to drive many of those solutions into rituals and constraints of some upstreams compared to others, so mileage varies. Some PRs are merged in minutes, others stay open for years :\

But certainly, OI and OO rock, even if my improvements have to be bolted on the side :)

**Comment by balrog-nona** [comment]
Hello everybody, I will introduce myself as well. I'm new in OI and I love contributing here. In fact, I'm new to the whole Unix/Linux world and with every contribution to OI I learn something new. So far, I do mainly basic updates but I hope that with time, I will put myself into some more sophisticated commits :)

**Comment by AssassinK786** [comment]
Hi, I'm Khush, I want to try out openindiana , but some of members who tried hae said it is not good for desktop os 
here are my specs:- https://linux-hardware.org/?probe=eff9350e7f
If it requires to work every alternative, don't you think that it should be improve the desktop usage and track why it is not improve
I am saying so because the work on storage, TCP/IP performance and bandwidth, AMP, virtualization, containers, cloud, reproducibility, scalability have worked prefectly just like linux but at same time linux community is trying to make that it is desktop.
why doesn't illumos/openindiana do, what is exact reason that the desktop env and package are quite limited

**Reply by AssassinK786** [reply]
@Toasterson @AndWac any comments

**Reply by Toasterson** [reply]
It is very hard to understand you. Many people have different use cases and we are few people. We make the use cases we have work. If you are interested in this work as a Hobby then welcome. If you want to have something running without investing your own work. Please go to the Hardware store and buy an of the shelf product.

If you have a suggestion for a topic to look into please post links to descriptions so we can read more in detail what you want to suggest us.

**Reply by AssassinK786** [reply]
@Toasterson What i'm trying to say is that 
illumos has been able to work on storage, TCP/IP performance and bandwidth, AMP, virtualization, containers, cloud, reproducibility, scalability prefectly like linux but the same effort is not done on desktop level which linux successfully managed to work 

So my question when we will be able to reach the level of liux desktop

**Comment by AndWac** [comment]
Companies want to make money. The areas you listed is where money can be earned with relatively low investments. Desktop is not that interesting because the margins are low (desktop users typically don't want to pay anything) and so you'll need to attract many. So, simply put, there is no working business plan to make money with desktop usage. Of course there are exceptions from this if you are able to attract many: Apple and Microsoft have build successful businesses with desktop users. But Microsoft nowadays has bigger interests in Azure than in Windows because you they make more money there with less developers.

Alas OpenIndiana is declining because we can only attract few people to get involved and even from this low numbers of maintainers only a fraction is interested in the continuation of Desktop usage.

**Comment by tsoome** [comment]
> On 6. Aug 2023, at 09:41, Khush Vasudeo Patil ***@***.***> wrote:
> 
> 
> @Toasterson <https://github.com/Toasterson> What i'm trying to say is that
> illumos has been able to work on storage, TCP/IP performance and bandwidth, AMP, virtualization, containers, cloud, reproducibility, scalability prefectly like linux but the same effort is not done on desktop level which linux successfully managed to work
> 
> So my question when we will be able to reach the level of liux desktop
> 
The unfortunate thing is, we do not have huge number of developers working to improve desktop, and therefore I do not think we have anything like concept what it is we want to have for desktop. So the answer is, it is when we get more developers to work to improve the desktop.

rgds,
toomas

**Comment by cruiztorresj** [comment]
Hello there all,

My name is [Caleb](https://www.calebjosue.com), I am a Software Developer located in Mexico.

Two weekends ago I've [installed ](https://www.calebjosue.com/blog.html#osoi)the latest version available of OpenIndiana, then I proceed to install [Racket ](https://racket-lang.org) from source. I want to contribute this package so new users arriving and trying to install this programming language can benefit from using the `pkg` command.

I wasn't able to subscribe to the dev mailing list so I am posting over here instead.

Is there anybody working on adding Racket to the package system? If not, I may give it a try in the upcoming weekends. any existing guide will be much appreciated.

Thanks!

**Comment by AndWac** [comment]
Hi Caleb,

Welcome to OpenIndiana! To my knowledge nobody is yet working on a port of Racket. If you want to do that, I recommend to look at existing ports (mainly their Makefiles) to find how similar problems have been solved. In the doc folder you can find lots of useful information, too. Furthermore we have an introduction on our web site, eg. https://docs.openindiana.org/dev/userland/

Regards,
Andreas
